### PR TITLE
Changes for PDB Downloader

### PR DESCRIPTION
**Pull request by GustavoLCR**
# PLEASE DO NOT SQUASH 

**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

* Remove unused `pdb.useragent`
* Make first parameter of `rz_bin_pdb_download()` be `RzBin`
* Implement `rz_bin_symserver_download()`

**Test plan**

Tests passs

**Closing issues**

None

### CODE REVIEW

File: librz/bin/pdb/pdb_downloader.c [hunk 1]
```diff
@@ -224,8 +211,26 @@ int rz_bin_pdb_download(RzCore *core, PJ *pj, int isradjson, SPDBOptions *option
 		rz_cons_printf("PDB \"%s\" download %s\n",
 			opt.dbg_file, ret ? "success" : "failed");
 	}
-	free(symbol_server);
 	free(path);
-
 	return !ret;
 }
+
+char *rz_bin_symserver_download(const SPDBDownloaderOpt *options) {
```
Comment (ret2libc) [inline librz/bin/pdb/pdb_downloader.c]: RZ_API

File: librz/bin/pdb/pdb_downloader.h [hunk 2]
```diff
@@ -49,7 +47,13 @@ void deinit_pdb_downloader(SPDBDownloader *pdb_downloader);
 /**
  * \brief download PDB file
  */
-RZ_API int rz_bin_pdb_download(RzCore *core, PJ *pj, int isradjson, SPDBOptions *options);
+RZ_API int rz_bin_pdb_download(RzBin *bin, PJ *pj, int isradjson, SPDBOptions *options);
+/**
```
Comment (ret2libc) [inline librz/bin/pdb/pdb_downloader.h]: not sure why these files have the doxygen doc here :( :( we usually track them in the .c files.

**Review by ret2libc — APPROVED** [review]

**Review by XVilka — APPROVED** [review]
Please address @ret2libc's comments and feel free to merge after.

File: librz/bin/pdb/pdb_downloader.c (hunk 1 above)
Comment (wargio) [inline librz/bin/pdb/pdb_downloader.c]:
```suggestion
RZ_API RZ_OWN char *rz_bin_symserver_download(RZ_NONNULL const SPDBDownloaderOpt *options) {
	rz_return_val_if(options, NULL);
```

File: librz/bin/pdb/pdb_downloader.c [hunk 3]
```diff
@@ -174,11 +172,10 @@ static bool is_valid_guid(const char *guid) {
 	return i >= 33; // len of GUID and age
 }
 
-int rz_bin_pdb_download(RzCore *core, PJ *pj, int isradjson, SPDBOptions *options) {
+int rz_bin_pdb_download(RzBin *bin, PJ *pj, int isradjson, SPDBOptions *options) {
 	int ret = 1;
```
Comment (wargio) [inline librz/bin/pdb/pdb_downloader.c:195]:
```suggestion
	rz_return_val_if(bin && options, 1);
	int ret = 1;
```

File: librz/bin/pdb/pdb_downloader.c [hunk 4]
```diff
@@ -174,11 +172,10 @@ static bool is_valid_guid(const char *guid) {
...
 	SPDBDownloaderOpt opt;
-	SPDBDownloader pdb_downloader;
-	RzBinInfo *info = rz_bin_get_info(core->bin);
+	RzBinInfo *info = rz_bin_get_info(bin);
 
 	if (!info || !info->debug_file_name) {
 		eprintf("Can't find debug filename\n");
```
Comment (wargio) [inline librz/bin/pdb/pdb_downloader.c]: RZ_LOG_ERROR

File: librz/bin/pdb/pdb_downloader.c [hunk 5]
```diff
@@ -190,29 +187,19 @@ int rz_bin_pdb_download(RzCore *core, PJ *pj, int isradjson, SPDBOptions *option
 		return 1;
 	}
 
-	if (!options || !options->symbol_server || !options->user_agent) {
+	if (!options || !options->symbol_server || !options->symbol_store_path) {
 		eprintf("Can't retrieve pdb configurations\n");
```
Comment (wargio) [inline librz/bin/pdb/pdb_downloader.c]: RZ_LOG_ERROR

File: librz/bin/pdb/pdb_downloader.c [hunk 6]
```diff
@@ -174,11 +172,10 @@ static bool is_valid_guid(const char *guid) {
 	return i >= 33; // len of GUID and age
 }
 
-int rz_bin_pdb_download(RzCore *core, PJ *pj, int isradjson, SPDBOptions *options) {
+int rz_bin_pdb_download(RzBin *bin, PJ *pj, int isradjson, SPDBOptions *options) {
```
Comment (wargio) [inline librz/bin/pdb/pdb_downloader.c]:
```suggestion
RZ_API int rz_bin_pdb_download(RZ_NONNULL RzBin *bin, PJ *pj, int isradjson, RZ_NONNULL SPDBOptions *options) {
```

File: librz/bin/pdb/pdb_downloader.h [hunk 7]
```diff
@@ -49,7 +47,13 @@ void deinit_pdb_downloader(SPDBDownloader *pdb_downloader);
 /**
  * \brief download PDB file
  */
-RZ_API int rz_bin_pdb_download(RzCore *core, PJ *pj, int isradjson, SPDBOptions *options);
+RZ_API int rz_bin_pdb_download(RzBin *bin, PJ *pj, int isradjson, SPDBOptions *options);
```
Comment (wargio) [inline librz/bin/pdb/pdb_downloader.h]:
```suggestion
RZ_API int rz_bin_pdb_download(RZ_NONNULL RzBin *bin, PJ *pj, int isradjson, RZ_NONNULL SPDBOptions *options);
```

File: librz/bin/pdb/pdb_downloader.h [hunk 8]
```diff
@@ -49,7 +47,13 @@ void deinit_pdb_downloader(SPDBDownloader *pdb_downloader);
...
+/**
+ * \brief downloads file from symbol server
+ * \param options options for downloading file
+ * \return char* is the path that file was downloaded to or NULL in case of failure
+ */
+RZ_API char *rz_bin_symserver_download(const SPDBDownloaderOpt *options);
```
Comment (wargio) [inline librz/bin/pdb/pdb_downloader.h]:
```suggestion
RZ_API RZ_OWN char *rz_bin_symserver_download(RZ_NONNULL const SPDBDownloaderOpt *options);
```

**Review by XVilka — APPROVED** [review]

**Review by wargio — APPROVED** [review]

**Pull request merged**